File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       int[] ans=new int[2];
       int i,j,n=nums.length;
       Map<Integer,Integer> map= new HashMap<>();
       for(i=0;i<n;i++){
            int t=(target-nums[i]);
            if(map.containsKey(t)){
                ans[0]=map.get(t);
                ans[1]=i;
                return ans;
            }else{
                map.put(nums[i],i);
            }
       }
       return ans;
        
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        n=len(nums)
        for i in range(0,n):
            c=target-nums[i]
            if c in map:
                return [map[c],i]
            map[nums[i]]=i
        return []